blosc2/blosc2_ext.pyx

```diff
@@ -2036,6 +2043,11 @@ cdef b2nd_context_t* create_b2nd_context(shape, chunks, blocks, dtype, kwargs):
     # This is used only in constructors, dtype will always have NumPy format
     dtype = np.dtype(dtype)
     typesize = dtype.itemsize
+    if 'cparams' in kwargs:
+        if 'typesize' not in kwargs['cparams']:
+            kwargs['cparams']['typesize'] = typesize
+    else:
+        kwargs['cparams'] = {'typesize': typesize}
```

Comment:
This won't override the type size if passed. We should changed the doc of some ndarray constructors.

Reply:
Good point.  Fixed in d5095eb